main/lsp/ExtractVariable.cc

```diff
@@ -268,4 +289,216 @@ vector<unique_ptr<TextDocumentEdit>> VariableExtractor::getEdits(LSPTypecheckerD
     res.emplace_back(move(docEdit));
     return res;
 }
+
+// This tree walk takes a ExpressionPtr and looks for nodes that are the same as that node
+class ExpressionPtrSearchWalk {
+    ast::ExpressionPtr *targetNode;
+    vector<const ast::ExpressionPtr *> enclosingScopeStack;
+    std::vector<core::LocOffsets> skippedLocs;
+
+    // NOTE: Might want to profile and switch to UnorderedSet.
+    bool shouldSkipLoc(core::LocOffsets loc) {
+        return absl::c_find_if(skippedLocs, [loc](auto l) { return l.contains(loc); }) != skippedLocs.end();
+    }
+
+public:
+    vector<pair<vector<const ast::ExpressionPtr *>, const ast::ExpressionPtr *>> matches;
```

Comment:
```suggestion
    vector<pair<vector<const ast::ExpressionPtr *>, core::LocOffsets>> matches;
```

Why are we storing an ExpressionPtr here? From my read of the code, we're only ever using `matches[N].second->loc()`.

In favor of storing as few raw pointers as possible, I'd love if we can make this change, assuming my read of the code is right.